spec/tasks/coa_upgrade/task_spec.rb

```diff
@@ -0,0 +1,108 @@
+require 'spec_helper'
+require 'tmpdir'
+require 'yaml'
+require_relative '../task_spec_helper'
+
+describe 'COA upgrade task' do
+  let(:yaml_files) { File.join('**', '*.yml') }
+  let(:stderr_path) { File.join(@upgrate_results, 'stderr.log') }
+  let(:stderr_loaded_file) do
+    content = File.read(stderr_path) if File.exist?(stderr_path)
+    content || ''
+  end
+
+  before(:context) do
+    @root_dir = File.expand_path(File.join(File.dirname(__FILE__), '..', '..', '..'))
+    reference_dataset = File.join(@root_dir, 'docs','reference_dataset')
+    reference_dataset_template = File.join(reference_dataset, 'template_repository')
+    reference_dataset_secrets = File.join(reference_dataset, 'config_repository')
+
+    @templates_dir =  Dir.mktmpdir
+    @config_dir = Dir.mktmpdir
+    @upgrated_templates_dir = Dir.mktmpdir
+    @upgrated_config_dir = Dir.mktmpdir
+    @upgrate_results = Dir.mktmpdir
+
+    FileUtils.cp_r(File.join(reference_dataset_template, '.'), @templates_dir)
+    FileUtils.cp_r(File.join(reference_dataset_secrets, '.'), @config_dir)
+    @fly_upgrade_cmd = '-c concourse/tasks/coa-upgrade/task.yml ' \
+        '-i cf-ops-automation=. ' \
+        "-i templates=#{@templates_dir} " \
+        "-i config=#{@config_dir} " \
+        "-o upgraded-config=#{@upgrated_config_dir} " \
+        "-o upgraded-templates=#{@upgrated_templates_dir} " \
+        "-o upgrade-results=#{@upgrate_results} "
+  end
+
+  after(:context) do
+    FileUtils.rm_rf @templates_dir
```

Comment:
The "official" pendant to `Dir.mktmpdir` is  `FileUtils.remove_entry_secure dir`. See https://ruby-doc.org/stdlib-2.3.1/libdoc/tmpdir/rdoc/Dir.html